### PR TITLE
CEPHSTORA-256:  fio benchmark code refactoring

### PR DESCRIPTION
**Pull request by srkreddy**
First pass of refactoring benchmark code.
This should help running specific benchmarks and create only required devices in the rados pool.

### CODE REVIEW

File: benchmark/fio_benchmark.yml [hunk 1]
```diff
@@ -1,4 +1,13 @@
 ---
+- name: Run pre-check
+  hosts:
+    - localhost
```
Comment (mvollman) [inline benchmark/fio_benchmark.yml]: Can we run this task as a local_action?  We recently were asked to remove localhost from the rpc-ceph playbooks

File: benchmark/fio_bench/fio_benchmark_setup.yml [hunk 2]
```diff
@@ -90,41 +90,69 @@
...
+        - name: Create a ceph device for benchmarking
+          command: "rbd {{ _ceph_args }} create {{ _pool_name }}/{{ item }} -s {{ _size }}"
+          with_items:
+            - "bench_file-{{ inventory_hostname_short }}"
+          register: create_result
+          failed_when: create_result.rc not in [0,17]
+          changed_when: create_result.rc == 0
+        - name: Map nbd file device
+          command: "rbd-nbd {{ _ceph_args }} map {{ _pool_name }}/bench_file-{{ inventory_hostname_short }}"
+          when: "'bench_file-' + inventory_hostname_short not in nbd_mapped_before.stdout"
```
Comment (mvollman) [inline benchmark/fio_bench/fio_benchmark_setup.yml]: This is a conflict.  nbd_mapped_before only exists if bench_direct_EBS is true

File: benchmark/fio_bench/fio_benchmark_setup.yml [hunk 3]
```diff
@@ -90,48 +90,83 @@
...
+          filesystem:
+            fstype: xfs
+            device: "{{ bench_file_device_name }}"
+        - name: Mount image
+          mount:
+            src: "{{ bench_file_device_name }}"
+            fstype: "xfs"
+            name: "/mnt/fio_bench"
+            state: mounted
+      when: (ebs_file_bench|default(false)) or (osa_file_bench|default(false))
```
Comment (mvollman) [inline benchmark/fio_bench/fio_benchmark_setup.yml]: Move to benchmark_hosts defaults.  Same as above.  See below.

File: benchmark/fio_benchmark.yml [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
+  tasks:
+    - name: fail when none of the benchmarks are chosen
+      fail:
+        msg: "Set atleast one of the benchmark options to true.
+              Options are: ebs_direct_bench, ebs_file_bench, osa_file_bench, ebs_rbd_bench, osa_rbd_bench"
+      when: not ((ebs_direct_bench | default(false)) or (ebs_file_bench | default(false)) or (osa_file_bench | default(false)) or
+                 (ebs_rbd_bench | default(false)) or (osa_rbd_bench | default(false)))
```
Comment (mvollman) [inline benchmark/fio_benchmark.yml]: Move to benchmark_hosts defaults.  Same as above.  See below.

File: benchmark/fio_bench/fio_benchmark_setup.yml [hunk 5]
```diff
@@ -90,48 +90,83 @@
...
+        bench_direct_device_name: ""
+    - block:
+        - name: Create a ceph device for benchmarking
+          command: "rbd {{ _ceph_args }} create {{ _pool_name }}/{{ item }} -s {{ _size }}"
+          with_items:
+            - "bench_direct_RBD-{{ inventory_hostname_short }}"
+          register: create_result
+          failed_when: create_result.rc not in [0,17]
+          changed_when: create_result.rc == 0
+      when: (ebs_rbd_bench|default(false)) or (osa_rbd_bench|default(false))
```
Comment (mvollman) [inline benchmark/fio_bench/fio_benchmark_setup.yml]: These variables should be defaulted in the group_vars/benchmark_hosts/00-defaults.yml so we are not assigning these variables defaults in multiple places. Please see comment below.

File: benchmark/fio_bench/fio_benchmark_setup.yml [hunk 6]
```diff
@@ -90,48 +90,83 @@
...
+          when: "'bench_direct_EBS-' + inventory_hostname_short not in nbd_mapped_before.stdout"
+        - name: Get mapped rbd-nbd devices
+          command: "rbd-nbd {{ _ceph_args }} list-mapped"
+          register: nbd_mapped_after
+        - name: Set bench_direct_device_name
+          set_fact:
+            bench_direct_device_name: "{{ nbd_mapped_after.stdout | regex_search(regexp, '\\1') | join('') }}"
+          vars:
+            regexp: 'bench_direct_EBS-{{ inventory_hostname_short }}.*(\/dev\/nbd[0-9]+)'
+      when: ebs_direct_bench|default(false)
```
Comment (mvollman) [inline benchmark/fio_bench/fio_benchmark_setup.yml]: Move to benchmark_hosts defaults.  Same as above.  See below.

File: playbooks/group_vars/benchmark_hosts/00-defaults.yml [hunk 7]
```diff
@@ -69,7 +69,7 @@ fio_bench_list_default:
     rw: write
     iodepth: 8
     numjobs: 1
-    run_bench: "{{ ebs_rbd_bench | default(true) }}"
+    run_bench: "{{ ebs_rbd_bench | default(false) }}"
```
Comment (mvollman) [inline playbooks/group_vars/benchmark_hosts/00-defaults.yml]: If we define these variables once in this file we will not have to give them defaults in multiple places.

I think we should set at the top of this file:
```
ebs_rbd_bench: false
```

And then set
```
    run_bench: "{{ ebs_rbd_bench }}"
```

We should do this for all the run_bench variables so we are not giving them default values in multiple playbooks.

Comment (srkreddy) [reply playbooks/group_vars/benchmark_hosts/00-defaults.yml]: addressing comments in next pr

**Pull request merged**